card/src/main/java/com/adyen/checkout/card/DefaultCardDelegate.kt

```diff
@@ -512,16 +512,45 @@ internal class DefaultCardDelegate(
         countryOptions: List<AddressListItem>,
         stateOptions: List<AddressListItem>
     ): AddressOutputData {
+        val isOptional = when (componentParams.addressParams) {
```

Comment:
This when logic could be part of another function. Let's keep the delegate clean and extract these segments to somewhere else. You can put this and `isAddressOptional()` in `CardValidationUtils` for now.